etc/adminhtml/system.xml

```diff
@@ -214,6 +214,14 @@
                     </comment>
                     <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                 </field>
+                <field id="customer_reference" translate="label comment" type="select" sortOrder="100"
```

Comment:
Check the sort order, 100 is already used. While you're at it, I'd fix all of these: https://github.com/Nosto/nosto-magento2/blob/5404057b5ccf0f86123ed0b18b1be8d6657852c2/etc/adminhtml/system.xml#L192-L224